src/main/java/io/horizen/lambo/car/transaction/BuyCarTransaction.java

```diff
@@ -1,180 +1,174 @@
+import com.horizen.box.data.NoncedBoxData;
+import com.horizen.box.data.RegularBoxData;
+import io.horizen.lambo.car.info.CarBuyOrderInfo;
+import com.horizen.proof.Proof;
+import com.horizen.proof.Signature25519;
+import com.horizen.proposition.Proposition;
+import com.horizen.transaction.TransactionSerializer;
+import com.horizen.transaction.AbstractRegularTransaction;
+import com.horizen.utils.BytesUtils;
+import scorex.core.NodeViewModifier$;
+
+import java.io.ByteArrayOutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static io.horizen.lambo.car.transaction.CarRegistryTransactionsIdsEnum.BuyCarTransactionId;
+
+// BuyCarTransaction is nested from AbstractRegularTransaction so support regular coins transmission as well.
+// BuyCarTransaction was designed to accept the SellOrder by specific buyer or to cancel it by the owner.
+// As outputs it contains possible RegularBoxes(to pay to the sell order owner and make change) and new CarBox entry.
+// As unlockers it contains RegularBoxes and CarSellOrder to open.
+public final class BuyCarTransaction extends AbstractRegularTransaction {
+
+    // CarBuyOrderInfo is a view that describes what sell order to open and who will be the next owner.
+    // But inside it contains just a minimum set of info (like CarSellOrderBox itself and proof) that is the unique source of data.
+    // So, no one outside controls what will be the specific outputs of this transaction.
+    // Any malicious actions will lead to transaction invalidation.
+    // For example, if SellOrder was accepted by the buyer specified, CarBuyOrderInfo view returns as the new box data
+    // new instance of CarBoxData the owned by the buyer and RegularBoxData with the payment to previous owner.
+    private final CarBuyOrderInfo carBuyOrderInfo;
+
+    public BuyCarTransaction(List<byte[]> inputRegularBoxIds,
+                             List<Signature25519> inputRegularBoxProofs,
+                             List<RegularBoxData> outputRegularBoxesData,
+                             CarBuyOrderInfo carBuyOrderInfo,
+                             long fee,
+                             long timestamp) {
+        super(inputRegularBoxIds, inputRegularBoxProofs, outputRegularBoxesData, fee, timestamp);
+
+        // Parameters sanity check
+        if(carBuyOrderInfo == null){
+            throw new IllegalArgumentException("Unacceptable value of carBuyOrderInfo!");
+        }
+        this.carBuyOrderInfo = carBuyOrderInfo;
+    }
+
+    // Specify the unique custom transaction id.
+    @Override
+    public byte transactionTypeId() {
+        return BuyCarTransactionId.id();
+    }
+
+    @Override
+    public List<NoncedBoxData<Proposition, NoncedBox<Proposition>>> getOutputData(){
+        List<NoncedBoxData<Proposition, NoncedBox<Proposition>>> allBoxesData = new ArrayList<>();
+        for(RegularBoxData regularBoxData: outputRegularBoxesData){

```

Comment:
I would prefer to have outputRegularBoxesData processed in AbstractRegularTransaction::getOutputData method, then call super method here.